src/Time/StepChoosers/RelaxTimeStepper.hpp

```diff
@@ -0,0 +1,84 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cmath>
+#include <limits>
+#include <pup.h>
+
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Time/StepChoosers/StepChooser.hpp"  // IWYU pragma: keep
+#include "Utilities/Registration.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class Time;
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+namespace Tags {
+struct HistoryEvolvedVariables;
+struct SubstepTime;
+}  // namespace Tags
+/// \endcond
+
+namespace StepChoosers {
+template <typename StepChooserRegistrars>
+class RelaxTimeStepper;
+
+namespace Registrars {
+using RelaxTimeStepper =
+    Registration::Registrar<StepChoosers::RelaxTimeStepper>;
+}  // namespace Registrars
+
+/// Avoids instabilities due to rapid increases in the step size by
+/// preventing the step size from increasing until the time stepper
+/// has recovered from previous changes.
+template <typename StepChooserRegistrars =
+              tmpl::list<Registrars::RelaxTimeStepper>>
+class RelaxTimeStepper : public StepChooser<StepChooserRegistrars> {
+ public:
+  /// \cond
+  RelaxTimeStepper() = default;
+  explicit RelaxTimeStepper(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(RelaxTimeStepper);  // NOLINT
+  /// \endcond
+
+  static constexpr OptionString help{
+      "Allows the time stepper to recover from previous changes."};
+  using options = tmpl::list<>;
+
+  using argument_tags =
+      tmpl::list<Tags::SubstepTime, Tags::HistoryEvolvedVariables>;
+
+  template <typename Metavariables, typename History>
+  double operator()(const Time& time, const History& history,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/)
+      const noexcept {
+    if (history.size() < 2) {
+      return std::numeric_limits<double>::infinity();
+    }
+
+    const auto last_step = time - history[history.size() - 1];
+
+    // Slab boundaries are complicated, so we'll just ignore the slab
+    // information and assume slab sizes don't change too much.
+    for (auto step = history.begin(); step != history.end() - 1; ++step) {
```

Comment:
Could you expand the C++ comment and also in the documentation that slab size changes are not accounted for because this looks only at the fractional changes of the time steps?